image/descriptor.go

```diff
@@ -105,7 +96,7 @@ func (d *descriptor) validate(w walker, mts []string) error {
 		return fmt.Errorf("invalid descriptor MediaType %q", d.MediaType)
 	}
 
-	parsed, err := digest.Parse(d.Digest)
+	parsed, err := digest.Parse(string(d.Digest))
```

Comment:
No need to parse a digest that is already a digest. Just call `d.Digest.Validate()`.